src/stringProcessing/precedenceException.js

```diff
@@ -18,14 +18,14 @@ var cannotBeBetweenAuxiliaryAndParticipleSpanish = require( "../researches/spani
  */
 module.exports = function( sentencePart, participleIndex ) {
 	var precedenceExceptionRegex;
-	switch ( this.constructor.name ) {
-		case "FrenchParticiple":
+	switch ( this.getLanguage() ) {
```

Comment:
The `this` is still being used here. While not being bound to any particular scope. I'd really want to make sure this is correct by passing along the language and executing `this.getLanguage` in the proper scope (Participle).